multi-agent/lib/mas/scheduling/models.py

```
@@ -0,0 +1,87 @@
+    cron_expression: Optional[str] = None
+    overlap_policy: ScheduleOverlapPolicy = ScheduleOverlapPolicy.SKIP
+    timezone: str = "UTC"
+    start_at: Optional[datetime] = None
+    end_at: Optional[datetime] = None
+    remaining_actions: Optional[int] = None
+    enabled: bool = True
+
+    @model_validator(mode="after")
+    def _exactly_one_trigger(self) -> "ScheduleDefinition":
+        if not self.interval and not self.cron_expression:
+            raise ValueError("Either interval or cron_expression is required")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)